src/Plotly.NET/ChartAPI/Chart2D.fs

```diff
@@ -6014,4 +6014,530 @@ module Chart2D =
+        /// <param name="TextPosition">Sets the position of text associated with each datum</param>
+        /// <param name="MultiTextPosition">Sets the position of text associated with individual datum</param>
+        /// <param name="MarkerColor">Sets the color of the marker for the Y values.</param>
+        /// <param name="MarkerColorScale">Sets the colorscale of the marker for the Y values.</param>
+        /// <param name="MarkerOutline">Sets the outline of the marker for the Y values.</param>
+        /// <param name="MarkerSymbol">Sets the marker symbol for each datum for the Y values.</param>
+        /// <param name="MultiMarkerSymbol">Sets the marker symbol for each individual datum for the Y values.</param>
+        /// <param name="Marker">Sets the marker (use this for more finegrained control than the other marker-associated arguments) for the Y values.</param>
+        /// <param name="MarkerSize">Sets the size of the datapoint markers for the Y values.</param>
+        /// <param name="LineColor">Sets the color of the line for the Y values.</param>
+        /// <param name="LineColorScale">Sets the colorscale of the line for the Y values.</param>
+        /// <param name="LineWidth">Sets the width of the line for the Y values.</param>
+        /// <param name="LineDash">sets the drawing style of the line for the Y values.</param>
+        /// <param name="Line">Sets the line (use this for more finegrained control than the other line-associated arguments) for the Y values.</param>
+        /// <param name="ShowReference">Sets if the reference Line should be visible.</param>
+        /// <param name="ReferenceLine">Sets the line (use this for more finegrained control than the other line-associated arguments) for the reference Y values.</param>
+        /// <param name="ReferenceWidth">Sets the width of the line for the reference Y values.</param>
+        /// <param name="ReferenceColor">Sets the color of the line for the reference Y values.</param>
+        /// <param name="ReferenceText">Sets a text associated with each datum for the reference Y values.</param>
+        /// <param name="MultiReferenceText">Sets individual text for each datum for the reference Y values.</param>
+        /// <param name="ReferenceTextPosition">Sets the position of text associated with each reference datum.</param>
+        /// <param name="MultiReferenceTextPosition">Sets the position of text associated with each individual reference datum</param>
+        /// <param name="TextFont">Sets the font used for `text` in Text.</param>
+        /// <param name="UseDefaults">If set to false, ignore the global default settings set in `Defaults`</param>
+        [<Extension>]
+        static member Residual
+            (
+                x: seq<#IConvertible>,
+                y: seq<#IConvertible>,
+                reference: seq<#IConvertible>,
+                ?Name: string,
+                ?ReferenceName: string,
+                ?GroupName: string,
```

Comment:
I think grouping the legend items here is unnecessary, as we will have one trace per group per default, and also same name for reference line and group:

<img width="231" height="157" alt="grafik" src="https://github.com/user-attachments/assets/33b940ba-3547-431c-b86d-ec4af00e1e48" />

please refactor so that, if no group names are set by the user, no trace grouping is done.